### PR TITLE
Add IdentityServer events to AdapterFactory

### PR DESCRIPTION
**Pull request by RaheelRsk**

### CODE REVIEW

**Review by rehcra22 — APPROVED** [review]
:shipit:

**Review by SundasChoudyRsk — CHANGES_REQUESTED** [review]

File: src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/UnhandledExceptionEventAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public class UnhandledExceptionEventAdapter : IAuditEventArguments
+    {
+        private readonly UnhandledExceptionEvent evt;
+
+        public UnhandledExceptionEventAdapter(UnhandledExceptionEvent evt)
+        {
+            this.evt = evt ?? throw new ArgumentNullException(nameof(evt));
+        }
+
+        public ResourceActor Actor => new ResourceActor(ResourceActor.MachineSubjectType, evt.ActivityId, evt.Details);
```
Comment (SundasChoudyRsk) [inline src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/UnhandledExceptionEventAdapter.cs:16]: Here, I don't there's any actor. Can we set actor identifier and displayName as null or an empty string maybe? I don't think the event has any properties on it that could determine the actor.

File: src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenRevokedSuccessEventAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        private readonly TokenRevokedSuccessEvent evt;
+
+        public TokenRevokedSuccessEventAdapter(TokenRevokedSuccessEvent evt)
+        {
+            this.evt = evt ?? throw new ArgumentNullException(nameof(evt));
+        }
+
+        public ResourceActor Actor => new ResourceActor(ResourceActor.UserSubjectType, evt.ClientId, evt.ClientName);
+        public string Action => evt.Name;
+        public AuditableResource Resource => new AuditableResource("IdentityServer", evt.ClientId);
```
Comment (SundasChoudyRsk) [inline src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenRevokedSuccessEventAdapter.cs:18]: The event has a property called `TokenType`, we can set the AuditableResource Identifier to this

File: src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionSuccessEventAdapter.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        private readonly TokenIntrospectionSuccessEvent evt;
+
+        public TokenIntrospectionSuccessEventAdapter(TokenIntrospectionSuccessEvent evt)
+        {
+            this.evt = evt ?? throw new ArgumentNullException(nameof(evt));
+        }
+
+        public ResourceActor Actor => new ResourceActor(ResourceActor.UserSubjectType, evt.ActivityId, evt.ApiName);
+        public string Action => evt.Name;
+        public AuditableResource Resource => new AuditableResource("IdentityServer", evt.ActivityId);
```
Comment (SundasChoudyRsk) [inline src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionSuccessEventAdapter.cs:18]: Here, the resource AuditableResource identifier can be `evt.ApiName`

File: src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionSuccessEventAdapter.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    public class TokenIntrospectionSuccessEventAdapter : IAuditEventArguments
+    {
+        private readonly TokenIntrospectionSuccessEvent evt;
+
+        public TokenIntrospectionSuccessEventAdapter(TokenIntrospectionSuccessEvent evt)
+        {
+            this.evt = evt ?? throw new ArgumentNullException(nameof(evt));
+        }
+
+        public ResourceActor Actor => new ResourceActor(ResourceActor.UserSubjectType, evt.ActivityId, evt.ApiName);
```
Comment (SundasChoudyRsk) [inline src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionSuccessEventAdapter.cs:16]: Here, I don't think we have a Resource Actor identifier. Should this be null or empty?

File: src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionFailureEventAdapter.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        private readonly TokenIntrospectionFailureEvent evt;
+
+        public TokenIntrospectionFailureEventAdapter(TokenIntrospectionFailureEvent evt)
+        {
+            this.evt = evt ?? throw new ArgumentNullException(nameof(evt));
+        }
+
+        public ResourceActor Actor => new ResourceActor(ResourceActor.UserSubjectType, evt.ActivityId, evt.ApiName);
+        public string Action => evt.Name;
+        public AuditableResource Resource => new AuditableResource("IdentityServer", evt.ActivityId);
```
Comment (SundasChoudyRsk) [inline src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionFailureEventAdapter.cs:18]: The AuditableResource should be `evt.ApiName`

File: src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionFailureEventAdapter.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    public class TokenIntrospectionFailureEventAdapter : IAuditEventArguments
+    {
+        private readonly TokenIntrospectionFailureEvent evt;
+
+        public TokenIntrospectionFailureEventAdapter(TokenIntrospectionFailureEvent evt)
+        {
+            this.evt = evt ?? throw new ArgumentNullException(nameof(evt));
+        }
+
+        public ResourceActor Actor => new ResourceActor(ResourceActor.UserSubjectType, evt.ActivityId, evt.ApiName);
```
Comment (SundasChoudyRsk) [inline src/Rsk.DuendeIdentityServer.AuditEventSink/Adapters/TokenIntrospectionFailureEventAdapter.cs:16]: Again here, there's not subject

**Review by SundasChoudyRsk — APPROVED** [review]

**Pull request merged**